code/game/machinery/atmoalter/gas_mine.dm

```diff
@@ -183,4 +183,27 @@
 
 /obj/machinery/atmospherics/miner/gas_giant/AddAir()
 	if(ticker)
-		air_contents.copy_from(gas_giant.GM)
\ No newline at end of file
+		air_contents.copy_from(gas_giant.GM)
+
+
+/obj/machinery/atmospherics/miner/mixed_nitrogen
+	name = "\improper Mixed Gas Miner"
+	desc = "Pumping nitrogen, carbon dioxide, and plasma."
+	overlay_color = "#FF80BD"
+
+/obj/machinery/atmospherics/miner/mixed_nitrogen/AddAir()
+	air_contents.adjust_multi(
```

Comment:
for readability, I suggest just having
```
var/total_adjust = internal_pressure * air_contents.volume / (R_IDEAL_GAS_EQUATION * air_contents.temperature)
air_contents.adjust_multi(GAS_CARBON, 0.3*total_adjust,
GAS_NITROGEN, .04*total_adjust,
etc etc
```

Reply:
actually, even better if you make it call a new proc `var/total_adjust = AirRate()` which returns the above formula.

Then AddAir() is just one line and it's really easy to customize your mixed gases on a map by map basis.

Reply:
Sounds like a good idea, will get to adding it like this soon

Reply:
when

Reply:
I think I have it now